packages/core/src/tools/fetch-logger.js

```diff
@@ -0,0 +1,106 @@
+const _ = require('lodash');
+
+const { ALLOWED_HTTP_DATA_CONTENT_TYPES } = require('./http');
+
+const stringifyRequestData = (data) => {
+  if (typeof data === 'string') {
+    return data;
+  } else if (data instanceof URLSearchParams) {
+    return data.toString();
+  } else if (global.FormData && data instanceof global.FormData) {
+    return '<FormData>';
+  } else {
+    // See https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Using_Fetch#setting_a_body
+    // for other possible body types
+    return '<unsupported format>';
+  }
+};
+
+const normalizeRequestInfo = (input, init) => {
+  const result = {
+    method: 'GET',
+    headers: {},
+    data: '',
+  };
+
+  if (global.Request && input instanceof global.Request) {
+    result.url = input.url;
+    result.method = input.method || result.method;
+    result.headers =
+      Object.fromEntries(input.headers.entries()) || result.headers;
+    if (input.body) {
+      result.data = stringifyRequestData(input.body);
+    }
+  } else if (typeof input.toString === 'function') {
+    // This condition includes `typeof input === 'string'` as well
+    result.url = input.toString();
+  } else {
+    // Don't log if we don't recognize the input type
+    return null;
+  }
+
+  if (init) {
+    result.method = init.method || result.method;
+    result.headers = init.headers || result.headers;
+    if (init.body) {
+      result.data = stringifyRequestData(init.body);
+    }
+  }
+
+  return result;
+};
+
+// Is this request made by z.request()?
+const isZapierUserAgent = (headers) =>
+  _.get(headers, 'user-agent', []).indexOf('Zapier') !== -1;
+
+const shouldIncludeResponseContent = (contentType) => {
+  for (const ctype of ALLOWED_HTTP_DATA_CONTENT_TYPES) {
+    if (contentType.includes(ctype)) {
+      return true;
+    }
+  }
+  return false;
+};
+
+const stringifyResponseContent = async (response) => {
+  // Be careful not to consume the original response body, which is why we clone it
+  return response.clone().text();
+};
+
+// Usage:
+//   global.fetch = wrapFetchWithLogger(global.fetch, logger);
+const wrapFetchWithLogger = (fetchFunc, logger) => {
+  if (fetchFunc.patchedByZapier) {
+    return fetchFunc;
+  }
+
+  const newFetch = async function (input, init) {
+    const response = await fetchFunc(input, init);
+    const requestInfo = normalizeRequestInfo(input, init);
+    if (requestInfo && !isZapierUserAgent(requestInfo.headers)) {
+      const responseContentType = response.headers.get('content-type');
+
+      logger(`${response.status} ${requestInfo.method} ${requestInfo.url}`, {
+        log_type: 'http',
+        request_type: 'patched-devplatform-outbound',
+        request_url: requestInfo.url,
+        request_method: requestInfo.method,
+        request_headers: requestInfo.headers,
+        request_data: requestInfo.data,
```

Comment:
@eliangcs Should we consider avoiding `streaming data` here? I recall streaming data for the patched `http` module being a likely cause of runaway ingress data increase, see [this Slack message](https://app.slack.com/client/T024VA8T9/C5Z9BP4U9) for context.

Reply:
It seems like `stringifyRequestData` handles this; just making sure to call this out explicitly just in case it was not considered 😄

Reply:
That's a good call. I [added](https://github.com/zapier/zapier-platform/pull/908/commits/82b50d5e0963ba852876cd94fff6e8d2a830a6e8) a comment in `stringifyRequestData` to call it out for posterity.